tests/lms/views/decorators/test_h_api.py

```diff
@@ -21,9 +16,39 @@
 
 @pytest.mark.usefixtures("hapi_svc")
 class TestCreateHUser:
+    def test_it_invokes_patch_for_user_update(
+        self, create_h_user, context, pyramid_request, hapi_svc
+    ):
+
+        create_h_user(pyramid_request, mock.sentinel.jwt, context)
+
+        hapi_svc.patch.assert_called_once_with(
+            f"users/{context.h_username}", {"display_name": context.h_display_name}
+        )
+
+    def test_it_continues_to_post_if_patch_raises_HAPINotFoundError(
+        self, create_h_user, context, pyramid_request, hapi_svc
+    ):
+        # It will only invoke POST if PATCH raises HAPINotFoundError
+        hapi_svc.patch.side_effect = HAPINotFoundError("whatever")
+
+        create_h_user(pyramid_request, mock.sentinel.jwt, context)
+
+        assert hapi_svc.post.call_count == 1
+
+    def test_it_raises_if_patch_raises_unexpected_error(
+        self, create_h_user, context, pyramid_request, hapi_svc
+    ):
+        hapi_svc.patch.side_effect = ValueError("whatever")
```

Comment:
I think this should be `HAPIError`. `patch()` doesn't raise `ValueError`. `test_it_raises_if_updating_the_user_fails()` might be a better test name (contains less implementation details) too